### ISSUE TITLE
LazyLoad for Iframes: Preview image cannot receive focus and fails the a11y audit

### ISSUE THREAD

**Issue by WordPresseur** [wp-media/wp-rocket#4436]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**
It appears the lazy load preview image can not receive focus, and even if hacked with a tabindex="0", pressing enter on the image doesn't load the video.

We don't have indeed the "highlighting" focus option and `tab` is even skipping the video. The original YT video has it, so it works if we bypass the optimization.

This fails a11y audits, it's an extremely high priority for us to get fixed. All American websites using your plugin can not legally use this feature.

**To Reproduce**
1. Add YouTube video with standard WP block
2. Enable the LL for YT with preview images in WPR
3. Visit the page and use `tab` to navigate between the elements.
4. The video will be skipped

**Expected behavior**
Our LL should preserve these accessibility attributes
 
**Additional context**
Ticket: https://secure.helpscout.net/conversation/1661178330/301214?folderId=1213662
Slack: https://wp-media.slack.com/archives/C08N8J6VC/p1634216849074000

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by michaelbourne** [comment]
Thanks for making this issue for me.

The simple solution is that you need to use a button element to load the Youtube video (in place of your `.play` div, perhaps?)

This would solve your accessibility issues as it relates to navigation, although the content of what that button does likely needs to be given with an `aria-label` like "click here to load this video". You could of course style your button as needed, such as the default bg image for the play button.

The hard solution involves you adding a `tabindex` and `role` to the JS generated `div` from your inline page javascript. From there, you need an event listener for the enter key being pressed, as `onclick` does not work with the entre key on a `div` element.

Areas of concern:
https://github.com/wp-media/wp-rocket/blob/develop/inc/Dependencies/RocketLazyload/Iframe.php#L173
https://github.com/wp-media/wp-rocket/blob/develop/inc/Dependencies/RocketLazyload/Assets.php#L237
